class.jetpack-geo-locate.php

```diff
@@ -0,0 +1,344 @@
+<?php
+
+class Jetpack_Geo_Locate {
+	private static $instance;
+
+	static public function init() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new Jetpack_Geo_Locate;
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * This is mostly just used for testing purposes.
+	 */
+	static public function reset_instance() {
+		self::$instance = null;
+	}
+
+	private function __construct() {
+		add_action( 'init', array( $this, 'wordpress_init' ) );
+		add_action( 'wp_head', array( $this, 'wp_head' ) );
+		add_filter( 'the_content', array( $this, 'the_content_microformat' ) );
+		add_filter( 'the_content', array( $this, 'the_content_location_display' ), 15, 1 );
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+
+		$this->register_rss_hooks();
+	}
+
+	/**
+	 * Register support for the geo-location feature on pages and posts.  Register the meta
+	 * fields managed by this plugin so that they are properly sanitized during save.
+	 */
+	public function wordpress_init() {
+		add_post_type_support( 'post', 'geo-location' );
+		add_post_type_support( 'page', 'geo-location' );
+
+		register_meta(
+			'post',
+			'geo_public',
+			array(
+				'sanitize_callback' => array( $this, 'sanitize_public' ),
+				'type'              => 'string',
+				'single'            => true,
+			)
+		);
+
+		register_meta(
+			'post',
+			'geo_latitude',
+			array(
+				'sanitize_callback' => array( $this, 'sanitize_coordinate' ),
+				'type'              => 'float',
+				'single'            => true,
+			)
+		);
+
+		register_meta(
+			'post',
+			'geo_longitude',
+			array(
+				'sanitize_callback' => array( $this, 'sanitize_coordinate' ),
+				'type'              => 'float',
+				'single'            => true,
+			)
+		);
+
+		register_meta(
+			'post',
+			'geo_address',
+			array(
+				'sanitize_callback' => 'sanitize_text_field',
+				'type'              => 'string',
+				'single'            => true,
+			)
+		);
+	}
+
+	/**
+	 * Filter "public" input to always be either 1 or 0.
+	 *
+	 * @param mixed $public
+	 *
+	 * @return int
+	 */
+	public function sanitize_public( $public ) {
+		return absint( $public ) ? 1 : 0;
+	}
+
+	/**
+	 * Filter geo coordinates and normalize them to floats with 7 digits of precision.
+	 *
+	 * @param mixed $coordinate
+	 *
+	 * @return float|null
+	 */
+	public function sanitize_coordinate( $coordinate ) {
+		if ( ! $coordinate ) {
+			return null;
+		}
+
+		return round( (float) $coordinate, 7 );
+	}
+
+	/**
+	 * Render geo.position and ICBM meta tags with public geo meta values when rendering
+	 * a single post.
+	 */
+	function wp_head() {
+		if ( ! $this->is_single() ) {
+			return;
+		}
+
+		global $post;
+
+		$meta_values = $this->get_meta_values( $post->ID );
+
+		if ( ! $meta_values['is_public'] ) {
+			return;
+		}
+
+		echo  "\n<!-- Jetpack Geo-location Tags -->\n";
+
+		printf(
+			'<meta name="geo.position" content="%s:%s" />' . PHP_EOL,
```

Comment:
Everything I found suggests that the delimiter between lat and long should be a `;` here, not `:`.

Reply:
I confirmed that this is supposed to be a `;` and corrected the error in 6c7458a.